railgun/test/integration/knative_test.go

```diff
@@ -0,0 +1,257 @@
+			}
+		}
+
+		t.Logf("All knative pods are up and ready.")
+		return true
+
+	}, 60*time.Second, 1*time.Second, true)
+}
+
+func TestKnativeIngress(t *testing.T) {
+	_ = proxyReady()
+	ctx := context.Background()
+
+	t.Log("Deploying all resources that are required to run knative")
+	err := deployManifest(knativeCrds, ctx, t)
+	assert.NoError(t, err)
+	err = deployManifest(knativeCore, ctx, t)
+	assert.NoError(t, err)
+	knativeReady := isKnativeReady(ctx, cluster, t)
+	assert.Equal(t, knativeReady, true)
+
+	t.Log("Note down the ip address or public CNAME of kong-proxy service.")
+	proxy := retrieveProxyInfo(ctx, t)
+	assert.NotEmpty(t, proxy)
+	if err != nil {
+		t.Fatalf("kong-proxy service ip/public name is not ready.")
+	}
+
+	t.Log("Configure Knative NetworkLayer as Kong")
+	err = configKnativeNetwork(ctx, cluster, t)
+	assert.NoError(t, err)
+	err = configKnativeDomain(ctx, proxy, cluster, t)
+	assert.NoError(t, err)
+
+	t.Log("Install knative service")
+	srvInstall := assert.Eventually(t, func() bool {
+		err = installKnativeSrv(ctx, t)
+		if err != nil {
+			t.Logf("checking knativing webhook readiness.")
+			return false
+		}
+		return true
+	}, 30*time.Second, 2*time.Second, true)
+	assert.EqualValues(t, true, srvInstall)
+
+	t.Log("Test knative service using kong.")
+	srvaccessable := accessKnativeSrv(ctx, proxy, t)
+	assert.EqualValues(t, true, srvaccessable)
+}
+
+func deployManifest(yml string, ctx context.Context, t *testing.T) error {
+	cmd := exec.CommandContext(ctx, "kubectl", "apply", "-f", yml)
+	stdout, stderr := new(bytes.Buffer), new(bytes.Buffer)
+	cmd.Stdout = stdout
+	cmd.Stderr = stderr
+	if err := cmd.Run(); err != nil {
+		fmt.Fprintln(os.Stdout, stdout.String())
+		return err
+	}
+	t.Logf("successfully deploy manifest " + yml)
+	return nil
+}
+
+func checkIPAddress(ip string, t *testing.T) bool {
+	if net.ParseIP(ip) == nil {
+		t.Logf("IP Address: %s - Invalid\n", ip)
+		return false
+	} else {
+		t.Logf("IP Address: %s - Valid\n", ip)
+		return true
+	}
+}
+
+func retrieveProxyInfo(ctx context.Context, t *testing.T) string {
+	var proxy string
+	assert.Eventually(t, func() bool {
+		cmd := exec.CommandContext(ctx, "kubectl", "get", "service", "ingress-controller-kong-proxy", "--namespace", "kong-system")
+		stdout, stderr := new(bytes.Buffer), new(bytes.Buffer)
+		cmd.Stdout = stdout
+		cmd.Stderr = stderr
+		if err := cmd.Run(); err != nil {
+			fmt.Fprintln(os.Stdout, stdout.String())
+			fmt.Fprintln(os.Stderr, stderr.String())
+			return false
+		}
+
+		if len(stdout.String()) > 0 {
+			info := strings.Split(stdout.String(), "\n")
+			proxy = strings.Fields(info[1])[3]
+			t.Logf("kong-proxy %s", proxy)
+			if checkIPAddress(proxy, t) == true {
+				return true
+			}
+		}
+		return false
+	}, 60*time.Second, 1*time.Second, true)
+	return proxy
+}
+
+func configKnativeNetwork(ctx context.Context, cluster kind.Cluster, t *testing.T) error {
+	payloadBytes := []byte("{\"data\": {\"ingress.class\": \"kong\"}}")
```

Comment:
We should use the dynamic ingress.class value from the test suite here:

```suggestion
	payloadBytes := []byte(fmt.Sprintf("{\"data\": {\"ingress.class\": \"%s\"}}", ingressClass))
```

see: https://github.com/Kong/kubernetes-ingress-controller/blob/main/railgun/test/integration/suite_test.go#L54